api/v1beta1/awscluster_webhook.go

```diff
@@ -114,6 +114,12 @@ func (r *AWSCluster) ValidateUpdate(old runtime.Object) error {
 					r.Spec.ControlPlaneLoadBalancer.Name, "field is immutable"),
 			)
 		}
+		if oldC.Spec.ControlPlaneLoadBalancer.HealthCheckProtocol.String() != ClassicELBProtocolSSL.String() && newLoadBalancer.HealthCheckProtocol != nil {
```

Comment:
instead of `oldC.Spec.ControlPlaneLoadBalancer` use `existingLoadBalancer` variable. In [Line#97 ](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta1/awscluster_webhook.go#L97)we are creating deepcopy for `oldC.Spec.ControlPlaneLoadBalancer`

Reply:
Done